browser/playlist/playlist_tab_helper.cc

```diff
@@ -258,19 +258,20 @@ void PlaylistTabHelper::UpdateSavedItemFromCurrentContents() {
   // perf improvement? We'll see this really matters.
 
   bool should_notify = false;
-  base::ranges::for_each(
-      service_->GetAllPlaylistItems(),
-      [this, &should_notify](const auto& item) {
-        const auto& current_url = web_contents()->GetVisibleURL();
-        if (item->page_source != current_url) {
-          return;
-        }
-
-        DVLOG(2) << __FUNCTION__ << " " << item->page_source.spec() << " "
-                 << item->media_source.spec();
-        saved_items_.push_back(item->Clone());
-        should_notify = true;
-      });
+  base::ranges::for_each(service_->GetAllPlaylistItems(),
+                         [this, &should_notify](const auto& item) {
+                           const auto& current_url =
+                               web_contents()->GetVisibleURL().GetWithoutRef();
+                           if (item->page_source != current_url) {
```

Comment:
Nice! I'm wondering if we could drop the ref of page_source here and compare it with current_url. If we can, we can close the PR in the playlist-component repo and keep the fragment data. I'm hoping we can keep the fragment part so that we make the link to the original page more accurate(which is not implemented yet).

Reply:
I've done the changes. So, we can close the PR in `playlist component` repo.